openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,1564 @@
+
+                    template = '{0: <12}Group {1} [{2: <10} - {3: <10}MeV]:\t'
+
+                    # Loop over energy groups ranges
+                    for group in range(1, self.num_groups+1):
+                        bounds = self.energy_groups.get_group_bounds(group)
+                        string += template.format('', group, bounds[0], bounds[1])
+                        average = self.get_xs([group], [subdomain], [nuclide],
+                                              xs_type=xs_type, value='mean',
+                                              delayed_groups=[delayed_group])
+                        rel_err = self.get_xs([group], [subdomain], [nuclide],
+                                              xs_type=xs_type, value='rel_err',
+                                              delayed_groups=[delayed_group])
+                        average = average.flatten()[0]
+                        rel_err = rel_err.flatten()[0] * 100.
+                        string += '{:.2e} +/- {:1.2e}%'.format(average, rel_err)
+                        string += '\n'
+                    string += '\n'
+                string += '\n'
+
+        print(string)
+
+    def export_xs_data(self, filename='mgxs', directory='mgxs',
+                       format='csv', groups='all', xs_type='macro',
+                       delayed_groups='all'):
+        """Export the multi-delayed-group cross section data to a file.
+
+        This method leverages the functionality in the Pandas library to export
+        the multi-group cross section data in a variety of output file formats
+        for storage and/or post-processing.
+
+        Parameters
+        ----------
+        filename : str
+            Filename for the exported file. Defaults to 'mgxs'.
+        directory : str
+            Directory for the exported file. Defaults to 'mgxs'.
+        format : {'csv', 'excel', 'pickle', 'latex'}
+            The format for the exported data file. Defaults to 'csv'.
+        groups : Iterable of Integral or 'all'
+            Energy groups of interest. Defaults to 'all'.
+        xs_type: {'macro', 'micro'}
+            Store the macro or micro cross section in units of cm^-1 or barns.
+            Defaults to 'macro'.
+        delayed_groups : Iterable of Integral or 'all'
+            Delayed groups of interest. Defaults to 'all'.
+
+        """
+
+        cv.check_type('filename', filename, basestring)
+        cv.check_type('directory', directory, basestring)
+        cv.check_value('format', format, ['csv', 'excel', 'pickle', 'latex'])
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+
+        # Make directory if it does not exist
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        filename = os.path.join(directory, filename)
+        filename = filename.replace(' ', '-')
+
+        # Get a Pandas DataFrame for the data
+        df = self.get_pandas_dataframe(groups=groups, xs_type=xs_type,
```

Comment:
I'll look into splitting things up to reduce redundant code since I know there is a fair amount.